### ISSUE TITLE
.htmlhintrc parsing issue (strict JSON)

### ISSUE THREAD

**Issue by Hufschmidt** [yaniswang/HTMLHint#94]
It seems like parsing of .htmlhintrc is broken to some degree:

# Issue:
    {
      "doctype-first": false
    }

With this, htmlhint stops reporting any issues, eg. no warning about *< h2 >Whoops< h2 >*. 
(I'm using this to lint AngularJS templates)

    {
      "doctype-first": false,
    }

Using this works just fine BUT this config can't be used by gulp-htmlhint (for deployment linting as cordova-hook) because .htmlhintrc can't be parsed.

Obviously the later is not valid (strict) JSON, so I guess there is an issue with htmlhint when parsing/loading this file, since the first config should be valid. (*htmlhint: 0.9.12*)

**Comment by yaniswang** [comment]
https://www.npmjs.com/package/strip-json-comments

May be is issue from `strip-json-comments`?

**Comment by Hufschmidt** [comment]
Tested you idea:
I removed strip-json-comments in ``/usr/lib/node_modules/htmlhint/gin/htmlhint`` (where the config JSON is parsed), but still the same problem.

Maybe there is a misconception on my part: 
 * When I don't use a htmlhintrc config-file, htmlhint will use its default set of rules
 * When I use a htmlhintrc config-file, htmlhint will ...
   * My assumption was: ... use its default ruleset with options from given config applied
   * What seems more likely: ... only use the options from given config and ignore its default ruleset

Which one is true?

**Comment by yaniswang** [comment]
Your are useing gulp-htmlhint ?

**Comment by Hufschmidt** [comment]
**TL;DR:** Yes, I'm using htmlhint with gulp-html and also via atom-linter-htmlhint. (See **Addendum**)

**Longer Version:**
Yes, I'm using htmlhint together with gulp (via gulp-htmlhint), so a bit more background details here. 

The following code works with gulp-htmlhint AND only disables the ``doctype-first`` rule:

    {
      "doctype-first": false
    }

However this does not (seem to) work with atom-linter-htmlhint, since I don't get any warning/errors in Atom anymore. I confirmed this is not a atom-linter-htmlhint issue, since manually running htmlhint on files with intentional errors also does not report any warning/error.

~~On the otherhand with the following code atom-linter-htmlhint continues to display all warnings/errors, but gulp-htmlhint fails, because it can't parse the JSON.~~

    {
      "doctype-first": false,
    }

**Addendum:** 

My bad, this actually *'works'*, because atom-linter-htmlhint (unlike gulp-htlmhint) fails silently when loading the above htmlhintrc, which causes it to be ignored...

This only leaves me with one question then:
Is it the intended behavior that htmlhint ignores it default ruleset when using a htmlhintrc config-file (aka disabling all rules not explicitly enabled in the config)?

**Comment by yaniswang** [comment]
Yes, the default rule will work when the config is undefined.

**Comment by Hufschmidt** [comment]
In that case my issue if resolved, or at least explained, thank you :)
